zkvm/src/prover.rs

```diff
@@ -60,11 +60,15 @@ impl<'a, 'b> Delegate<r1cs::Prover<'a, 'b>> for Prover<'a, 'b> {
 }
 
 impl<'a, 'b> Prover<'a, 'b> {
-    pub fn build_tx<'g>(
+    pub fn build_tx<'g, F>(
         program: Vec<Instruction>,
         header: TxHeader,
         bp_gens: &'g BulletproofGens,
-    ) -> Result<(Tx, TxID, TxLog), VMError> {
+        sign_tx_fn: F,
```

Comment:
Note for later: we'll most likely have a whole trait/interface/type called `Signer` or something, and it'll support  multiple keys, multiple parties and all of the above at various levels. Cf. https://github.com/interstellar/slingshot/issues/146

Reply:
With such interface it'd make sense to compose all these pubkeys into a N-of-N "witness" object (yes, on the verifier's side) which would be the same witness object that provers would use for their custom N-of-N multisig rules packed in a single key (not visible to the verifier).

Reply:
sounds good, will make a note!